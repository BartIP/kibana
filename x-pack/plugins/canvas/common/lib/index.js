/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export * from './datatable';
<<<<<<< HEAD
export * from './arg';
export * from './ast';
=======
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
export * from './autocomplete';
export * from './constants';
export * from './dataurl';
export * from './errors';
export * from './expression_form_handlers';
export * from './fetch';
export * from './find_in_object';
<<<<<<< HEAD
export * from './fn';
export * from './fonts';
export * from './functions_registry';
export * from './get_by_alias';
export * from './get_colors_from_palette';
export * from './get_field_type';
export * from './get_legend_config';
export * from './get_type';
export * from './grammar';
=======
export * from './fonts';
export * from './get_colors_from_palette';
export * from './get_field_type';
export * from './get_legend_config';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
export * from './handlebars';
export * from './hex_to_rgb';
export * from './httpurl';
export * from './latest_change';
export * from './missing_asset';
export * from './palettes';
export * from './pivot_object_array';
<<<<<<< HEAD
export * from './registry';
export * from './resolve_dataurl';
export * from './serialize';
export * from './type';
export * from './types_registry';
=======
export * from './resolve_dataurl';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
export * from './unquote_string';
export * from './url';
