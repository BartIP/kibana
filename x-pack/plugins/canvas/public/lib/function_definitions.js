/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import uniqBy from 'lodash.uniqby';
<<<<<<< HEAD
import { getServerFunctions } from '../state/selectors/app';
import { getBrowserRegistries } from './browser_registries';
=======
import { getBrowserRegistries } from '@kbn/interpreter/public/browser_registries';
import { getServerFunctions } from '../state/selectors/app';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

export async function getFunctionDefinitions(state) {
  const { browserFunctions } = await getBrowserRegistries();
  const serverFunctions = getServerFunctions(state);
  return uniqBy(serverFunctions.concat(browserFunctions.toArray()), 'name');
}
