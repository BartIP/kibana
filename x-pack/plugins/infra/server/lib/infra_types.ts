/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { InfraConfigurationAdapter } from './adapters/configuration';
import { InfraBackendFrameworkAdapter, InfraFrameworkRequest } from './adapters/framework';
<<<<<<< HEAD
import { InfraCapabilitiesDomain } from './domains/capabilities_domain';
import { InfraFieldsDomain } from './domains/fields_domain';
import { InfraLogEntriesDomain } from './domains/log_entries_domain';
=======
import { InfraFieldsDomain } from './domains/fields_domain';
import { InfraLogEntriesDomain } from './domains/log_entries_domain';
import { InfraMetadataDomain } from './domains/metadata_domain';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { InfraMetricsDomain } from './domains/metrics_domain';
import { InfraNodesDomain } from './domains/nodes_domain';
import { InfraSourceStatus } from './source_status';
import { InfraSourceConfigurations, InfraSources } from './sources';

export interface InfraDomainLibs {
<<<<<<< HEAD
  capabilities: InfraCapabilitiesDomain;
=======
  metadata: InfraMetadataDomain;
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  fields: InfraFieldsDomain;
  logEntries: InfraLogEntriesDomain;
  nodes: InfraNodesDomain;
  metrics: InfraMetricsDomain;
}

export interface InfraBackendLibs extends InfraDomainLibs {
  configuration: InfraConfigurationAdapter<InfraConfiguration>;
  framework: InfraBackendFrameworkAdapter;
  sources: InfraSources;
  sourceStatus: InfraSourceStatus;
}

export interface InfraConfiguration {
  enabled: boolean;
  query: {
    partitionSize: number;
    partitionFactor: number;
  };
  sources: InfraSourceConfigurations;
}

export interface InfraContext {
  req: InfraFrameworkRequest;
}
