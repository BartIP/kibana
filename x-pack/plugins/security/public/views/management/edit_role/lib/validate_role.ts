/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

/*
 * Copyright Elasticsearch B.V. ../../../../../common/model/index_privileger one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

<<<<<<< HEAD
=======
import { i18n } from '@kbn/i18n';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { IndexPrivilege } from '../../../../../common/model/index_privilege';
import { KibanaPrivilege } from '../../../../../common/model/kibana_privilege';
import { Role } from '../../../../../common/model/role';

/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

interface RoleValidatorOptions {
  shouldValidate?: boolean;
}

export interface RoleValidationResult {
  isInvalid: boolean;
  error?: string;
}

export class RoleValidator {
  private shouldValidate?: boolean;

  private inProgressSpacePrivileges: any[] = [];

  constructor(options: RoleValidatorOptions = {}) {
    this.shouldValidate = options.shouldValidate;
  }

  public enableValidation() {
    this.shouldValidate = true;
  }

  public disableValidation() {
    this.shouldValidate = false;
  }

  public validateRoleName(role: Role): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    if (!role.name) {
<<<<<<< HEAD
      return invalid(`Please provide a role name`);
    }
    if (role.name.length > 1024) {
      return invalid(`Name must not exceed 1024 characters`);
    }
    if (!role.name.match(/^[a-zA-Z_][a-zA-Z0-9_@\-\$\.]*$/)) {
      return invalid(
        `Name must begin with a letter or underscore and contain only letters, underscores, and numbers.`
=======
      return invalid(
        i18n.translate(
          'xpack.security.management.editRoles.validateRole.provideRoleNameWarningMessage',
          {
            defaultMessage: 'Please provide a role name',
          }
        )
      );
    }
    if (role.name.length > 1024) {
      return invalid(
        i18n.translate(
          'xpack.security.management.editRoles.validateRole.nameLengthWarningMessage',
          {
            defaultMessage: 'Name must not exceed 1024 characters',
          }
        )
      );
    }
    if (!role.name.match(/^[a-zA-Z_][a-zA-Z0-9_@\-\$\.]*$/)) {
      return invalid(
        i18n.translate(
          'xpack.security.management.editRoles.validateRole.nameAllowedCharactersWarningMessage',
          {
            defaultMessage:
              'Name must begin with a letter or underscore and contain only letters, underscores, and numbers.',
          }
        )
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      );
    }
    return valid();
  }

  public validateIndexPrivileges(role: Role): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    if (!Array.isArray(role.elasticsearch.indices)) {
<<<<<<< HEAD
      throw new TypeError(`Expected role.elasticsearch.indices to be an array`);
=======
      throw new TypeError(
        i18n.translate('xpack.security.management.editRoles.validateRole.indicesTypeErrorMessage', {
          defaultMessage: 'Expected {elasticIndices} to be an array',
          values: {
            elasticIndices: '"role.elasticsearch.indices"',
          },
        })
      );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
    }

    const areIndicesValid =
      role.elasticsearch.indices
        .map(indexPriv => this.validateIndexPrivilege(indexPriv))
        .find((result: RoleValidationResult) => result.isInvalid) == null;

    if (areIndicesValid) {
      return valid();
    }
    return invalid();
  }

  public validateIndexPrivilege(indexPrivilege: IndexPrivilege): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    if (indexPrivilege.names.length && !indexPrivilege.privileges.length) {
<<<<<<< HEAD
      return invalid(`At least one privilege is required`);
=======
      return invalid(
        i18n.translate(
          'xpack.security.management.editRoles.validateRole.onePrivilegeRequiredWarningMessage',
          {
            defaultMessage: 'At least one privilege is required',
          }
        )
      );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
    }
    return valid();
  }

  public validateSelectedSpaces(
    spaceIds: string[],
    privilege: KibanaPrivilege | null
  ): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    // If no assigned privilege, then no spaces are OK
    if (!privilege) {
      return valid();
    }

    if (Array.isArray(spaceIds) && spaceIds.length > 0) {
      return valid();
    }
<<<<<<< HEAD
    return invalid('At least one space is required');
=======
    return invalid(
      i18n.translate(
        'xpack.security.management.editRoles.validateRole.oneSpaceRequiredWarningMessage',
        {
          defaultMessage: 'At least one space is required',
        }
      )
    );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  }

  public validateSelectedPrivilege(
    spaceIds: string[],
    privilege: KibanaPrivilege | null
  ): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    // If no assigned spaces, then a missing privilege is OK
    if (!spaceIds || spaceIds.length === 0) {
      return valid();
    }

    if (privilege) {
      return valid();
    }
<<<<<<< HEAD
    return invalid('Privilege is required');
=======
    return invalid(
      i18n.translate(
        'xpack.security.management.editRoles.validateRole.privilegeRequiredWarningMessage',
        {
          defaultMessage: 'Privilege is required',
        }
      )
    );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  }

  public setInProgressSpacePrivileges(inProgressSpacePrivileges: any[]) {
    this.inProgressSpacePrivileges = [...inProgressSpacePrivileges];
  }

  public validateInProgressSpacePrivileges(role: Role): RoleValidationResult {
    const { global } = role.kibana;

    // A Global privilege of "all" will ignore all in progress privileges,
    // so the form should not block saving in this scenario.
    const shouldValidate = this.shouldValidate && !global.includes('all');

    if (!shouldValidate) {
      return valid();
    }

    const allInProgressValid = this.inProgressSpacePrivileges.every(({ spaces, privilege }) => {
      return (
        !this.validateSelectedSpaces(spaces, privilege).isInvalid &&
        !this.validateSelectedPrivilege(spaces, privilege).isInvalid
      );
    });

    if (allInProgressValid) {
      return valid();
    }
    return invalid();
  }

  public validateSpacePrivileges(role: Role): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    const privileges = Object.values(role.kibana.space || {});

    const arePrivilegesValid = privileges.every(assignedPrivilege => !!assignedPrivilege);
    const areInProgressPrivilegesValid = !this.validateInProgressSpacePrivileges(role).isInvalid;

    if (arePrivilegesValid && areInProgressPrivilegesValid) {
      return valid();
    }
    return invalid();
  }

  public validateForSave(role: Role): RoleValidationResult {
    const { isInvalid: isNameInvalid } = this.validateRoleName(role);
    const { isInvalid: areIndicesInvalid } = this.validateIndexPrivileges(role);
    const { isInvalid: areSpacePrivilegesInvalid } = this.validateSpacePrivileges(role);

    if (isNameInvalid || areIndicesInvalid || areSpacePrivilegesInvalid) {
      return invalid();
    }

    return valid();
  }
}

function invalid(error?: string): RoleValidationResult {
  return {
    isInvalid: true,
    error,
  };
}

function valid(): RoleValidationResult {
  return {
    isInvalid: false,
  };
}
