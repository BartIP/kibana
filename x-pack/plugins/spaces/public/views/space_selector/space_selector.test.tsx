/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

<<<<<<< HEAD
import { render, shallow } from 'enzyme';
import React from 'react';
=======
import React from 'react';
import { renderWithIntl, shallowWithIntl } from 'test_utils/enzyme_helpers';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import chrome from 'ui/chrome';
import { Space } from '../../../common/model/space';
import { SpacesManager } from '../../lib/spaces_manager';
import { SpaceSelector } from './space_selector';

function getHttpAgent(spaces: Space[] = []) {
  const httpAgent: any = () => {
    return;
  };
  httpAgent.get = jest.fn(() => Promise.resolve({ data: spaces }));

  return httpAgent;
}

function getSpacesManager(spaces: Space[] = []) {
  const manager = new SpacesManager(getHttpAgent(spaces), chrome, '/');

  const origGet = manager.getSpaces;
  manager.getSpaces = jest.fn(origGet);

  return manager;
}

test('it renders without crashing', () => {
  const spacesManager = getSpacesManager();
<<<<<<< HEAD
  const component = shallow(<SpaceSelector spaces={[]} spacesManager={spacesManager as any} />);
=======
  const component = shallowWithIntl(
    <SpaceSelector.WrappedComponent
      spaces={[]}
      spacesManager={spacesManager as any}
      intl={null as any}
    />
  );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  expect(component).toMatchSnapshot();
});

test('it uses the spaces on props, when provided', () => {
  const spacesManager = getSpacesManager();

  const spaces = [
    {
      id: 'space-1',
      name: 'Space 1',
      description: 'This is the first space',
    },
  ];

<<<<<<< HEAD
  const component = render(<SpaceSelector spaces={spaces} spacesManager={spacesManager as any} />);
=======
  const component = renderWithIntl(
    <SpaceSelector.WrappedComponent
      spaces={spaces}
      spacesManager={spacesManager as any}
      intl={null as any}
    />
  );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

  return Promise.resolve().then(() => {
    expect(component.find('.spaceCard')).toHaveLength(1);
    expect(spacesManager.getSpaces).toHaveBeenCalledTimes(0);
  });
});

test('it queries for spaces when not provided on props', () => {
  const spaces = [
    {
      id: 'space-1',
      name: 'Space 1',
      description: 'This is the first space',
    },
  ];

  const spacesManager = getSpacesManager(spaces);

<<<<<<< HEAD
  shallow(<SpaceSelector spacesManager={spacesManager as any} />);
=======
  shallowWithIntl(
    <SpaceSelector.WrappedComponent spacesManager={spacesManager as any} intl={null as any} />
  );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

  return Promise.resolve().then(() => {
    expect(spacesManager.getSpaces).toHaveBeenCalledTimes(1);
  });
});
