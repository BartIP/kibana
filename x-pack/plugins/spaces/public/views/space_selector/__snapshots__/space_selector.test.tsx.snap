// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders without crashing 1`] = `
<EuiPage
  className="spcSpaceSelector"
  data-test-subj="kibanaSpaceSelector"
  restrictWidth={false}
>
  <EuiPageBody
    restrictWidth={false}
  >
    <EuiPageHeader
      className="spcSpaceSelector__heading"
      responsive={true}
    >
      <EuiSpacer
        size="xxl"
      />
      <span
        className="spcSpaceSelector__logo"
      >
        <EuiIcon
          size="xxl"
          type="logoKibana"
        />
      </span>
      <EuiTitle
        size="l"
        textTransform="none"
      >
        <h1>
<<<<<<< HEAD
          Select your space
=======
          <FormattedMessage
            defaultMessage="Select your space"
            id="xpack.spaces.spaceSelector.selectSpacesTitle"
            values={Object {}}
          />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
        </h1>
      </EuiTitle>
      <EuiText
        color="subdued"
        grow={true}
        size="s"
      >
        <p>
<<<<<<< HEAD
          You can change your space at anytime
=======
          <FormattedMessage
            defaultMessage="You can change your space at anytime"
            id="xpack.spaces.spaceSelector.changeSpaceAnytimeAvailabilityText"
            values={Object {}}
          />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
        </p>
      </EuiText>
    </EuiPageHeader>
    <EuiPageContent
      className="spcSpaceSelector__pageContent"
      panelPaddingSize="l"
    >
      <EuiFlexGroup
        alignItems="center"
        component="div"
        direction="column"
        gutterSize="l"
        justifyContent="flexStart"
        responsive={false}
        wrap={false}
      />
      <EuiSpacer
        size="xl"
      />
      <SpaceCards
        onSpaceSelect={[Function]}
        spaces={Array []}
      />
      <React.Fragment>
        <EuiSpacer
          size="l"
        />
        <EuiText
          color="subdued"
          grow={true}
          size="m"
          textAlign="center"
        >
<<<<<<< HEAD
          No spaces match search criteria
=======
          <FormattedMessage
            defaultMessage="No spaces match search criteria"
            id="xpack.spaces.spaceSelector.noSpacesMatchSearchCriteriaDescription"
            values={Object {}}
          />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
        </EuiText>
      </React.Fragment>
    </EuiPageContent>
  </EuiPageBody>
</EuiPage>
`;
